### PR TITLE
Adding device  info to all Branch server requests

### PR DESCRIPTION
**Pull request by sojanpr**
Adds Device info to all request to Branch Server

@aaustin @derrickstaten @ahmednawar

### CODE REVIEW

File: Branch-SDK/Branch-SDK/BNCServerInterface.m [hunk 1]
```diff
@@ -228,4 +233,38 @@ - (BNCServerResponse *)processServerResponse:(NSURLResponse *)response data:(NSD
     return serverResponse;
 }
 
+- (void)updateDeviceInfoToMutableDictionary:(NSMutableDictionary *)dict {
+    BNCPreferenceHelper *preferenceHelper = [BNCPreferenceHelper preferenceHelper];
+    BOOL isRealHardwareId;
+    NSString *hardwareId = [BNCSystemObserver getUniqueHardwareId:&isRealHardwareId andIsDebug:preferenceHelper.isDebug];
+    if (hardwareId) {
+        dict[BRANCH_REQUEST_KEY_HARDWARE_ID] = hardwareId;
```
Comment (aaustin) [inline Branch-SDK/Branch-SDK/BNCServerInterface.m]: can you grab `ad_tracking_enabled` as well: https://github.com/BranchMetrics/iOS-Deferred-Deep-Linking-SDK/blob/master/Branch-SDK/Branch-SDK/Requests/BranchInstallRequest.m#L48

**Comment by aaustin** [comment]
looks good other than that comment :+1:

**Comment by aaustin** [comment]
this looks good :+1:

**Comment by ahmednawar** [comment]
I like this PR. :+1:

**Pull request merged**